File: src/Entity/Message.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Entity;

import org.eclipse.persistence.jpa.jpql.parser.DateTime;

/**
 *
 * @author oussema
 */
public class Message {
    int id_message;
    int from_user;
    int to_conv;
    String text;
    DateTime date_time;

    public Message() {
    }

    public Message(int id_message, int from_user, int to_conv, String text) {
        this.id_message = id_message;
        this.from_user = from_user;
        this.to_conv = to_conv;
        this.text = text;
    }

    public Message(int id_message, int from_user, int to_conv, String text, DateTime date_time) {
        this.id_message = id_message;
        this.from_user = from_user;
        this.to_conv = to_conv;
        this.text = text;
        this.date_time = date_time;
    }

    public int getId_message() {
        return id_message;
    }

    public void setId_message(int id_message) {
        this.id_message = id_message;
    }

    public int getFrom_user() {
        return from_user;
    }

    public void setFrom_user(int from_user) {
        this.from_user = from_user;
    }

    public int getTo_conv() {
        return to_conv;
    }

    public void setTo_conv(int to_conv) {
        this.to_conv = to_conv;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public DateTime getDate_time() {
        return date_time;
    }

    public void setDate_time(DateTime date_time) {
        this.date_time = date_time;
    }

    @Override
    public String toString() {
        return "Message{" + "id_message=" + id_message + ", from_user=" + from_user + ", to_conv=" + to_conv + ", text=" + text + ", date_time=" + date_time + '}';
    }
    
}

File: symfony/src/Entity/Message.php
<?php

namespace App\Entity;

use App\Repository\MessageRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MessageRepository::class)]
class Message
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column (name: "id_message")]
    private ?int $id = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name: 'from_user', referencedColumnName: 'id_user')]
    private ?User $from_user = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name: 'to_conv', referencedColumnName: 'id')]
    private ?Conv $to_conv = null;

    #[ORM\Column(length: 255)]
    private ?string $text = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $date_time = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getFromUser(): ?User
    {
        return $this->from_user;
    }

    public function setFromUser(?User $from_user): self
    {
        $this->from_user = $from_user;

        return $this;
    }

    public function getToConv(): ?Conv
    {
        return $this->to_conv;
    }

    public function setToConv(?Conv $to_conv): self
    {
        $this->to_conv = $to_conv;

        return $this;
    }

    public function getText(): ?string
    {
        return $this->text;
    }

    public function setText(string $text): self
    {
        $this->text = $text;

        return $this;
    }

    public function getDateTime(): ?\DateTimeInterface
    {
        return $this->date_time;
    }

    public function setDateTime(\DateTimeInterface $date_time): self
    {
        $this->date_time = $date_time;

        return $this;
    }
}
